File: javascript/src/Behavioral/Iterator/IteratorInterface.js
(function() {
  define([], function() {
    var IteratorInterface;
    return IteratorInterface = (function() {
      function IteratorInterface() {}

      IteratorInterface.prototype.first = function() {
        throw 'Unimplemented method, IteratorInterface.first';
      };

      IteratorInterface.prototype.next = function() {
        throw 'Unimplemented method, IteratorInterface.next';
      };

      IteratorInterface.prototype.isDone = function() {
        throw 'Unimplemented method, IteratorInterface.isDone';
      };

      IteratorInterface.prototype.currentItem = function() {
        throw 'Unimplemented method, IteratorInterface.currentItem';
      };

      return IteratorInterface;

    })();
  });

}).call(this);

File: python/src/Behavioral/Iterator/IteratorInterface.py



#/// Original copy of this content taken from http://www.fluffycat.com/Java-Design-Patterns/ in 2010
#/// Original Author: Larry Truett
#/// Privacy Policy at http://www.fluffycat.com/Privacy-Policy/
#/Iterator (aka Cursor) Overview
#/One object can traverse all of the elements of another object.
#/DvdListIterator.java - the Iterator Interface

define [],()->
	class IteratorInterface
		first : -> throw 'Unimplemented method, IteratorInterface.first'
		next : -> throw 'Unimplemented method, IteratorInterface.next'
		isDone : -> throw 'Unimplemented method, IteratorInterface.isDone'
		currentItem : -> throw 'Unimplemented method, IteratorInterface.currentItem'

